### DISCUSSION TITLE
Relaxed rules when using cocotb runners

### DISCUSSION THREAD

**Discussion by cameronweston** [nickg/nvc#928] [General]
With the release of cocotb 1.9.0 I've been testing out nvc and have been very happy with it so far. When compiling some legacy code that I'm using, I get an error that a shared variable needs to be protected. It then suggests to use the ```--relaxed``` option to downgrade this to a warning. I tried to do this but the build step in the cocotb runner places build args before the ```-a``` option when executing. 

Does anyone know if there is a way to use relaxed rules with cocotb? Or is this an issue that needs to be brought up in the cocotb project?

**Comment by nickg** [comment]
Are you using the Python or Makefile runner? For the Makefile runner I think you can just add it to `COMPILE_ARGS` but the Python runner does indeed only allow customisation through `build_args` which goes before the `-a` command (it sets the "global" options). Probably best to discuss this with the cocotb folks. @ktbarrett

**Comment by cameronweston** [comment]
I’ve been using python runners. Okay I’ll open an issue with them. Thanks!

**Reply by nickg** [reply]
If all your code is 93 compatible adding `--std=1993` to the build args may also work.

**Reply by cameronweston** [reply]
Unfortunately I'm using 2008 features in my newer code it's just this legacy stuff I'm stuck with.
